service/src/builder.rs

```diff
@@ -291,22 +199,110 @@ fn ensure_binary(package: &Package) -> anyhow::Result<()> {
 
 /// Make sure "cdylib" is set for shuttle-next projects, else set it if possible.
 fn ensure_cdylib(package: &Package) -> anyhow::Result<()> {
-    if package.targets().iter().any(|target| target.is_lib()) {
+    if package.targets.iter().any(is_cdylib) {
         Ok(())
     } else {
         bail!("Your Shuttle next project must be a library. Please add `[lib]` to your Cargo.toml file.")
     }
 }
 
-/// Ensure `panic = "abort"` is not set:
-fn check_no_panic(ws: &Workspace) -> anyhow::Result<()> {
-    if let Some(profiles) = ws.profiles() {
-        for profile in profiles.get_all().values() {
-            if profile.panic.as_deref() == Some("abort") {
-                return Err(anyhow!("Your Shuttle project cannot have panics that abort. Please ensure your Cargo.toml does not contain `panic = \"abort\"` for any profiles."));
-            }
+fn is_cdylib(target: &Target) -> bool {
+    target.kind.iter().any(|kind| kind == "cdylib")
+}
+
+fn compiler(
+    packages: Vec<&Package>,
+    release_mode: bool,
+    wasm: bool,
+    project_path: PathBuf,
+) -> anyhow::Result<Vec<BuiltService>> {
+    let jobs = std::thread::available_parallelism()?.get();
+    let manifest_path = project_path.join("Cargo.toml");
+
+    let mut cargo = std::process::Command::new("cargo");
```

Comment:
Is there any point to making the compiling async if `cargo` is going to be using all of the CPU's anyway?

Reply:
I think so. The compilation is going to use most of the CPU, but while its task can be blocked on IO (e.g. reading files, downloading crates) it can yield the execution to a task that is not blocked and make sure we extract the most out of the asynchronous runtime. This becomes more important when this part of the `service/src/builder` will be owned by a service which will handle the building for more than one project.